internal/pkg/agent/install/uninstall.go

```diff
@@ -112,9 +138,62 @@ func Uninstall(cfgFile, topPath, uninstallToken string, log *logp.Logger, pt *pr
 	}
 	pt.Describe("Removed install directory")
 
+	if notifyFleet {
+		notifyFleetAuditUninstall(ctx, log, pt, cfg, ai)
+	}
+
 	return nil
 }
 
+// notifyFleetAuditUninstall will attempt to notify fleet-server of the agent's uninstall.
+//
+// There are retries for the attempt after a 10s wait, but it is a best-effort approach.
+func notifyFleetAuditUninstall(ctx context.Context, log *logp.Logger, pt *progressbar.ProgressBar, cfg *configuration.Configuration, ai *info.AgentInfo) {
+	pt.Describe("notify fleet-server of uninstall")
+	client, err := fleetclient.NewAuthWithConfig(log, cfg.Fleet.AccessAPIKey, cfg.Fleet.Client)
+	if err != nil {
+		pt.Describe(fmt.Sprintf("notify fleet-server: unable to create fleetapi client: %v", err))
+		return
+	}
+	cmd := fleetapi.NewAuditUnenrollCmd(ai, client)
+	req := &fleetapi.AuditUnenrollRequest{
+		Reason:    fleetapi.ReasonUninstall,
+		Timestamp: time.Now().UTC(),
+	}
+	timer := time.NewTimer(0)
+	for i := 0; i < 10; i++ {
```

Comment:
The number of attempts and the retry interval should be injected from the outside as parameters.

Even if those values are not configurable right now it could be useful to inject them from the main uninstall function based on configuration or command line flags in the future and we can make the tests run faster by specifying smaller values.